kb/2025-03-25/cve-2025-1974.md

```diff
@@ -0,0 +1,65 @@
+---
+title: 'CVE-2025-1974: ingress-nginx admission controller RCE escalation'
+description: 'This article describes the mitigation steps for the CVE-2025-1974 vulnerability in Harvester.'
+authors:
+  - name: Ivan Sim
+    title: Principal Software Engineer
+    url: https://github.com/ihcsim
+    image_url: https://github.com/ihcsim.png
+tags: [security, cve]
+hide_table_of_contents: false
+---
+
+:::warning
+CVE-2025-1974 has been rated as critical with a score of 9.8 of 10 (CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H).
+:::
+
+A security issue was discovered in Kubernetes where under certain conditions, an unauthenticated attacker with access to the pod network can achieve arbitrary code execution in the context of the ingress-nginx controller. This can lead to disclosure of secrets accessible to the controller. (Note that in the default installation, the controller can access all secrets cluster-wide.)
+
+Your Harvester cluster is affected by this vulnerability if:
+
+- the RKE2 ingress-nginx controller is installed, and
+- the pods are running version v1.12.0-hardened6
```

Comment:
Does this also impact older versions before 1.12?

Reply:
Fixed in 37491d40d3410547b798cd871f3211ceaaed12fd.